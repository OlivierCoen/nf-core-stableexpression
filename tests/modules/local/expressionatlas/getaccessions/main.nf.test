nextflow_process {

    name "Test Process EXPRESSIONATLAS_GETACCESSIONS"
    script "modules/local/expressionatlas/getaccessions/main.nf"
    process "EXPRESSIONATLAS_GETACCESSIONS"
    tag "getaccessions"
    tag "module"

    test('["solanum_tuberosum", ["potato"]]') {

        tag "potato_two_kw"

        when {

            process {
                """
                input[0] = "solanum_tuberosum"
                input[1] = "potato"
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

    test('["solanum_tuberosum", "potato,phloem"]') {

        tag "potato_two_kw"

        when {

            process {
                """
                input[0] = "solanum_tuberosum"
                input[1] = "potato,phloem"
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

    test('["solanum_tuberosum", null]') {

        tag "potato_no_kw"

        when {

            process {
                """
                input[0] = "solanum_tuberosum"
                input[1] = ""
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
