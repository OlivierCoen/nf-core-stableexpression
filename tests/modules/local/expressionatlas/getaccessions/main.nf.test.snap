{
    "[\"solanum_tuberosum\", [null]": {
        "content": [
            {
                "0": [
                    "accessions.csv:md5,dad63ac7a1715277fa44567bc40b5872"
                ],
                "1": [
                    [
                        "EXPRESSIONATLAS_GETACCESSIONS",
                        "python",
                        "Python 3.13.0"
                    ]
                ],
                "2": [
                    [
                        "EXPRESSIONATLAS_GETACCESSIONS",
                        "requests",
                        "2.32.3"
                    ]
                ],
                "3": [
                    [
                        "EXPRESSIONATLAS_GETACCESSIONS",
                        "nltk",
                        "3.9.1"
                    ]
                ],
                "csv": [
                    "accessions.csv:md5,dad63ac7a1715277fa44567bc40b5872"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-10-17T00:26:46.488292622"
    },
    "[\"solanum_tuberosum\", [\"potato\", \"phloem\"]]": {
        "content": [
            {
                "0": [
                    "accessions.txt:md5,dad63ac7a1715277fa44567bc40b5872"
                ],
                "1": [
                    [
                        "EXPRESSIONATLAS_GETACCESSIONS",
                        "python",
                        "Python 3.13.0"
                    ]
                ],
                "2": [
                    [
                        "EXPRESSIONATLAS_GETACCESSIONS",
                        "requests",
                        "2.32.3"
                    ]
                ],
                "3": [
                    [
                        "EXPRESSIONATLAS_GETACCESSIONS",
                        "nltk",
                        "3.9.1"
                    ]
                ],
                "txt": [
                    "accessions.txt:md5,dad63ac7a1715277fa44567bc40b5872"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-11-26T05:57:18.83875097"
    },
    "[\"solanum_tuberosum\", [\"potato\", \"phloem\"]": {
        "content": [
            {
                "0": [
                    "accessions.csv:md5,dad63ac7a1715277fa44567bc40b5872"
                ],
                "1": [
                    [
                        "EXPRESSIONATLAS_GETACCESSIONS",
                        "python",
                        "Python 3.13.0"
                    ]
                ],
                "2": [
                    [
                        "EXPRESSIONATLAS_GETACCESSIONS",
                        "requests",
                        "2.32.3"
                    ]
                ],
                "3": [
                    [
                        "EXPRESSIONATLAS_GETACCESSIONS",
                        "nltk",
                        "3.9.1"
                    ]
                ],
                "csv": [
                    "accessions.csv:md5,dad63ac7a1715277fa44567bc40b5872"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-10-17T00:26:09.77988473"
    },
    "[\"solanum_tuberosum\", \"potato,phloem\"]": {
        "content": [
            {
                "0": [
                    "accessions.txt:md5,dad63ac7a1715277fa44567bc40b5872"
                ],
                "1": [
                    [
                        "EXPRESSIONATLAS_GETACCESSIONS",
                        "python",
                        "Python 3.13.0"
                    ]
                ],
                "2": [
                    [
                        "EXPRESSIONATLAS_GETACCESSIONS",
                        "requests",
                        "2.32.3"
                    ]
                ],
                "3": [
                    [
                        "EXPRESSIONATLAS_GETACCESSIONS",
                        "nltk",
                        "3.9.1"
                    ]
                ],
                "txt": [
                    "accessions.txt:md5,dad63ac7a1715277fa44567bc40b5872"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-11-26T06:07:40.84221728"
    },
    "[\"solanum_tuberosum\", null]": {
        "content": [
            {
                "0": [
                    "accessions.txt:md5,dad63ac7a1715277fa44567bc40b5872"
                ],
                "1": [
                    [
                        "EXPRESSIONATLAS_GETACCESSIONS",
                        "python",
                        "Python 3.13.0"
                    ]
                ],
                "2": [
                    [
                        "EXPRESSIONATLAS_GETACCESSIONS",
                        "requests",
                        "2.32.3"
                    ]
                ],
                "3": [
                    [
                        "EXPRESSIONATLAS_GETACCESSIONS",
                        "nltk",
                        "3.9.1"
                    ]
                ],
                "txt": [
                    "accessions.txt:md5,dad63ac7a1715277fa44567bc40b5872"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-11-26T05:57:46.770758821"
    },
    "[\"solanum_tuberosum\", [\"potato\"]]": {
        "content": [
            {
                "0": [
                    "accessions.txt:md5,dad63ac7a1715277fa44567bc40b5872"
                ],
                "1": [
                    [
                        "EXPRESSIONATLAS_GETACCESSIONS",
                        "python",
                        "Python 3.13.0"
                    ]
                ],
                "2": [
                    [
                        "EXPRESSIONATLAS_GETACCESSIONS",
                        "requests",
                        "2.32.3"
                    ]
                ],
                "3": [
                    [
                        "EXPRESSIONATLAS_GETACCESSIONS",
                        "nltk",
                        "3.9.1"
                    ]
                ],
                "txt": [
                    "accessions.txt:md5,dad63ac7a1715277fa44567bc40b5872"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-11-26T05:56:33.599076724"
    },
    "Should run without failures": {
        "content": [
            {
                "0": [
                    "accessions.csv:md5,dad63ac7a1715277fa44567bc40b5872"
                ],
                "accession": [
                    "accessions.csv:md5,dad63ac7a1715277fa44567bc40b5872"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-10-14T18:18:45.473497815"
    }
}